### PR TITLE
Improvements to automatic app restarting

### PR DESCRIPTION
**Pull request by alexmuller**
I haven't tested this in vagrant yet, so it's probably best to not merge it unless you're supremely confident it's ok.

### CODE REVIEW

File: modules/monitoring/files/usr/lib/nagios/plugins/reload_service [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+# Script that NRPE runs when it's notified by the monitoring machine that an application
+# has crossed the high memory threshold.
+#
+set -eu
+
+APPNAME=$1
+
+logger --tag govuk_icinga_event_handler "Reloading service ${APPNAME} because monitoring has detected it using too much memory"
+
+sudo /sbin/initctl reload ${APPNAME}
```
Comment (mattbostock) [inline modules/monitoring/files/usr/lib/nagios/plugins/reload_service]: I think the `sudo` here is unneeded, as it's the `nagios` user we're allowing in the sudoers config. I'll raise a PR to remove.

Comment (rjw1) [inline modules/monitoring/files/usr/lib/nagios/plugins/reload_service]: @mattbostock are you sure about that?

Comment (mattbostock) [inline modules/monitoring/files/usr/lib/nagios/plugins/reload_service]: @rjw1 Can you please be less cryptic?

Comment (rjw1) [inline modules/monitoring/files/usr/lib/nagios/plugins/reload_service]: how does it know to escalate privileges if you don't call it with sudo?

Comment (mattbostock) [inline modules/monitoring/files/usr/lib/nagios/plugins/reload_service]: Good point :-D Thanks @rjw1

**Pull request merged**